src/test/test_failure_slice.cpp

```
@@ -0,0 +1,285 @@
+    auto prog_or_error = unmarshal(raw_prog, options);
+    auto inst_seq = std::get_if<InstructionSeq>(&prog_or_error);
+    REQUIRE(inst_seq != nullptr);
+
+    const Program prog = Program::from_sequence(*inst_seq, raw_prog.info, options);
+    auto result = analyze(prog);
+
+    return result.compute_failure_slices(prog);
+}
+
+// Test that extract_instruction_deps correctly identifies register reads/writes
+TEST_CASE("extract_instruction_deps for Bin instruction", "[failure_slice][deps]") {
```

Clang (14.0.6)

[warning] variable 'TEST_CASE' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)